# Board Report

In [1]:
import matplotlib.pyplot as plt
import explore_

### 1. Which lesson appears to attract the most traffic consistently across cohorts (per program)?

In [2]:
# Load solution from victoire folder
result = explore_.get_explore_question_1().set_index("cohort_id")
result.sort_index()

program_id          lesson  referred_to_lesson_count
cohort_id                                                      
1.0               1.0    javascript-i                      1039
2.0               1.0         content                        57
4.0               1.0         prework                         2
6.0               1.0   javascript-ii                        11
7.0               1.0         content                       245
8.0               1.0        java-iii                       155
9.0               4.0         content                         4
11.0              1.0         content                        79
12.0              1.0         content                       258
13.0              1.0         content                      2254
14.0              1.0         content                      1128
15.0              2.0         content                       241
16.0              2.0          spring                       110
17.0              1.0         content                      1359
18.0              2.0    javascript-i                       187
19.0              1.0         content                       650
21.0              2.0          spring                      1408
22.0              2.0           mysql                      3787
23.0              2.0    javascript-i                      3747
24.0              2.0    javascript-i                      4117
25.0              2.0    javascript-i                      4074
26.0              2.0    javascript-i                      3567
27.0              2.0    javascript-i                      2797
28.0              2.0           mysql                      4780
29.0              2.0    javascript-i                      4963
31.0              2.0    javascript-i                      3561
32.0              2.0    javascript-i                      4745
33.0              2.0    javascript-i                      5991
34.0              3.0  1-fundamentals                      3124
51.0              2.0    javascript-i                      4142
52.0              2.0        html-css                      3675
53.0              2.0    javascript-i                      4382
55.0              3.0           3-sql                      2499
56.0              2.0    javascript-i                      3767
57.0              2.0    javascript-i                      4092
58.0              2.0    javascript-i                      4237
59.0              3.0  classification                      5029
61.0              2.0    javascript-i                      2530
62.0              2.0    javascript-i                      4438
132.0             2.0    javascript-i                      3418
133.0             3.0  classification                      2715
134.0             2.0    javascript-i                      2968
135.0             2.0    javascript-i                      4942
137.0             3.0    fundamentals                      3448
138.0             2.0    javascript-i                      2630
139.0             2.0    javascript-i                      1195

In [3]:
result.lesson.value_counts()

javascript-i      23
content           10
spring             2
mysql              2
classification     2
prework            1
javascript-ii      1
java-iii           1
html-css           1
1-fundamentals     1
3-sql              1
fundamentals       1
Name: lesson, dtype: int64

- Javascript-i attract the most traffic consistently across cohorts and programs.

### 2. Is there a cohort that referred to a lesson significantly more than other cohorts seemed to gloss over?

In [ ]:
plt.ioff()
# Load solution from victoire folder
explore_.get_explore_question_2()

- Ceres (id-33) and Marco (id-135) cohorts have referred to the javascript-1 lessons significantly more than others.

### 3. Are there students who, when active, hardly access the curriculum? If so, what information do you have about these students?

In [5]:
one_day_list, days_from_end_df = explore_.get_explore_question_3()

In [6]:
days_from_end_df

days_apart_from_end_date  user_count
1 days                    0.145007         106
2 days                    0.087551          64
3 days                    0.065663          48
135 days                  0.061560          45
4 days                    0.051984          38
...                            ...         ...
51 days                   0.001368           1
43 days                   0.001368           1
45 days                   0.001368           1
91 days                   0.001368           1
188 days                  0.001368           1

[119 rows x 2 columns]

In [7]:
one_day_list

date  path
user_id            
918         1     1
940         1     1
619         1     1
879         1     1
278         1     3
832         1     3
388         1     4
539         1     5
812         1     6
487         1     9
697         1    10
340         1    10
941         1    10
572         1    11

- The average user logged into the curriculum 73 times. So all users who averaged less were considered as 'hardly access'. Also, we took into account dates during their cohort completion time.

- There were 14 users that only logged into the curriculum one day out of their entire course

- 14% of the low activity users last logged onto the curriculum one day before graduation

- 6% of low activity users last logged in 135 days before graduation

- Noteworthy is the fourth spot with 45 users who's last login was 135 days before their graduation. Followed by the other notables: 48 days and 89 days with the 8th and 10th spot, respectively.

- 62 users had random days associated with their last login before graduation, ranging from 19 days all the way up to 183 days before graduation.

### 4. At some point in 2019, the ability for students and alumni to access both curriculums (web dev to ds, ds to web dev) should have been shut off. Do you see any evidence of that happening? Did it happen before?

- Web development students didn't access the data science curriculum in 2019, although data scientists did access the web development curriculum.

- It appears that the access for data scientists to see the web development curriculum was turned off about 27 Oct 2019

- The access was turned back on in November 2019 and saw significant traffic that month, and then it was turned on again in May and November 2020, at these dates some students engaged in minor traffic to the Web Development curriculum.

- It is impossible to determine if such access restrictions were made in the past given the information in these tables, because if there were data science cohorts before Aug 2018, they were not distinguished in the database by a cohort id

### 5. What topics are grads continuing to reference after graduation and into their jobs (for each program)?

**Topics that webdev students continue to access more than 4 months after their graduation**

Top 30% of all traffic, in order of most visited:
- javascript -i
- spring: general, setup
- html-css: general, elements
- java -iii: general, servlets, jsp-and-jstl
- java -ii
- appendix
- javascript -ii
- toc
- java -i
- mysql
- jquery: main, weather map
- spring fundamentals: repositories, views, controllers
- SQL: mysql overview

Every other page not on this list that was visited by a web dev alumni 4 months after their graduation represents .6% or less of all such traffic. 

**Topics that data science students continue to access more than 4 months after their graduation**
Top 30% of all traffic, in order of most visited:
- Classification: overview and 'scale-features-or-not'
- Data Science Fundamentals: intro to data science, git
- SQL overview
- Anomaly detection: overview
- Regression, model: split-and-scale, evaluate
- Classification: decision trees
- Clustering: wrangle, explore, model
- Timeseries modeling
- Python: introduction

Every other page not on this list that was visited by a data science alumni 4 months after their graduation represents .8% or less of all such traffic. 